src/extension/background.js

```diff
@@ -97,6 +97,12 @@ const messageHandlers = {
     images.forEach(name => delete customBackgrounds[name]);
     storeImages({ images: customBackgrounds });
   },
+  revertAll: (message, port) => {
+    log("revertAllThemes", message);
+    browser.theme.reset();
+    messageHandlers.setTheme(message);
```

Comment:
`message` looks like `{type: "revertAll"}` here.

`setTheme` does `const theme = normalizeTheme(message.theme);`, and since `theme` is not set, FirefoxColor's default theme is used. This is incorrect, the purpose of the revert functionality is to use Firefox's default theme.

`setTheme` calls `storeTheme` and `applyTheme`. Since you're already calling `browser.theme.reset()`, the `applyTheme` call is unnecessary. So what remains is the need to call `storeTheme({ theme: null })`.

Reply:
yes, I thought when this was last discussed it was decided to revert back to the default FF color theme; which is what this does.

I tried storeTheme ~:

```javascript
  browser.theme.reset();
  storeTheme({ theme: null })
```

and if you refresh the old theme is restored. let me know if I am misunderstanding your feedback though. I have to do more testing to be sure but I think the only way to keep the true default theme (the non FF color default theme), is to disable this extension or remove this default theme altogether.

Reply:
> and if you refresh the old theme is restored.

Refresh what? If I look at src/extension/background.js, I can see that no theme is supposed to apply at extension startup. If the theme changes, it might be the result of logic in the web page.

Reply:
> Refresh what?

if you refresh the page..

there is a default purplish theme that is applied when you first install this extension:

<img src="https://monosnap.com/image/ReK1Ey2CLEDs6uCucctk58MFDDSqhp.png width="300" />

it has always been this way as far as I know..

Reply:
Sorry, I should have been clearer. I wanted to ask when the theme is restored. I.e. if you click "Revert", whether that would restore the default, and stay at the default if you reload the extension **without an active Firefox Color webpage**.

When I install the Firefox Color add-on for the first time, then a new tab opens, and only then the Firefox Color **web page** changes the theme to the default.json theme.

I expect that with the patch, the Firefox Color **extension** will not change the theme after a restart. Is this the case?

Secondly, if you open the web page again, will it not touch Firefox's default theme? Based on your previous comment, it seems that the web page will change the default theme to default.json when the user visits the Firefox Color website. This should not happen.

Reply:
> if you click "Revert", whether that would restore the default, and stay at the default if you reload the extension without an active Firefox Color webpage.

I just want to make sure we are talking about the same things, so I'm just adding some notes to clarify what is meant by "default":

- "FF Default theme" = FF's Default theme

- "*Firefox Color* Default theme" = default.json from Firefox Color (i.e. purplish/gray color that is set when you initial activate this addon).

So I tested this locally by adding the addon via `about:debugging`, clicking random and then hitting revert. Then I removed the addon via `about:debugging`. I am not 100% this simulates a live version but based on this, it does appear to keep go back to the "FF Default Theme" when I remove this extension in the way I've described above.

If I do NOT remove it, it seems to keep the "*Firefox Color* Default theme," unless the theme is changed through the extension of course. 


> I expect that with the patch, the Firefox Color extension will not change the theme after a restart. Is this the case?

If the FF color extension is active it _should_ use the last theme displayed or the "*Firefox Color* Default theme" (default here would be displayed at the beginning or after hitting "Revert.") 

If it's not active, based on my testing noted above, it _should_ be the FF theme set under customize theme.


But again I am not 100% due to testing locally - do you have a suggestion on how to test this locally to be 100% sure? :)

Reply:
When an extension is removed, Firefox will forget about the theme sey by the (removed) extension, and consequently fall back to Firefox's default theme.

> If the FF color extension is active it should use the last theme displayed or the "Firefox Color Default theme" (default here would be displayed at the beginning or after hitting "Revert.")

When "Revert" is hit, Firefox's default theme should be used by Firefox Color.
After a browser restart, Firefox Color should use the theme before the browser/addon restart, which is Firefox's default theme.

To test the behavior of an add-on after a browser restart, you have to actually install an add-on instead of loading it as a temporary add-on via `about:debugging`/`web-ext run`.

1. Install Firefox Nightly.
2. Visit `about:config` and set `xpinstall.signatures.required` to `false`.
3. Install the xpi that is generated by `npm run xpi:dev` (this does not exist yet, but IIRC you were working on it in another bug).

In most cases, you don't need to fully restart the browser. Simply disabling then re-enabling the add-on is often a realistic test.

Reply:
Okay so currently as I have it here, it reverts back to the default.json (the starting "canvas"). I'll have to look into this more to see if we can keep the actual FF default theme once the user hits reset..

I'll get back to you on this :)

Also, thanks for the testing notes :)

Reply:
Hey @Rob--W 

I copied over the xpi:dev script since that PR is still pending but I have not been able to test this locally - installing the xpi and uninstalling. 

I guess I am realizing that since that PR is still pending, maybe I should have waited as it might still needs adjustments - but I wanted to verify this: that this should work locally as well?

Reply:
hmm it seems that build/extension/manifest.json enables and disables the extension locally. I wonder if this is also a sufficient way to test for local dev...?

Reply:
I recommended against `about:debugging` before because you wanted to test a browser restart, but the extension would be uninstalled due to it being temporary.

If you don't restart the browser, then it is OK to use `about:debugging` or `web-ext` + R (reload).

When you disabled signature checking (see my previous comment in this thread), then you don't need to use `about:debugging` or `web-ext` to reload the extension. You can simply open the `.xpi` file in Firefox to install it. This experience is much closer to what the user will do in practice.

Reply:
> To test the behavior of an add-on after a browser restart, you have to actually install an add-on...

how would i go about installing the addon w/out using about:debugging /addons page? I tried dragging the unsigned dev xpi to the browser but not sure if that's working since I still see "Get Firefox Color" button.

(note: I have set `xpi.signatures.required` to false)

Reply:
> how would i go about installing the addon w/out using about:debugging /addons page? I tried dragging the unsigned dev xpi to the browser

That should work.

> but not sure if that's working since I still see "Get Firefox Color" button.

Did you get an install prompt? If so, it did work, but it could be that you didn't install the right xpi file. Where did you get your xpi file?

For `localhost`, you should run `npm run package` and use the generated `addon.xpi`. You can open the directory as a `file:///`-URL in your browser and click on `addon.xpi` to install it. If this is not obvious (which I'm afraid is the case), consider adding it to the README in your existing attempt to improve README (this is optional, you don't have to do it).

Reply:
huh?! 
yeh I didn't know about `npm run package` (I just haven't needed it yet). I noticed after running this script an `addon.xpi` gets created which I saw gets added to the root of the project. I'll definitely update the README so it's more clear for others that are working on this project. thanks!